crux-core/src/crux/kv/tx_log.clj

```diff
@@ -95,9 +95,12 @@
                                           (cons (assoc (decode-tx-event-key-from k)
                                                        :crux.tx.event/tx-events (mem/<-nippy-buffer (kv/value iterator)))
                                                 (tx-log (kv/next iterator))))))]
-                               (->> (tx-log (kv/seek iterator (encode-tx-event-key-to nil {::tx/tx-id (or after-tx-id 0)})))
-                                    (take batch-size)
-                                    vec)))]
+                               (let [first-key (kv/seek iterator (encode-tx-event-key-to nil {::tx/tx-id (or after-tx-id 0)}))]
```

Comment:
In hindsight, might be able to simplify this to `{::tx/tx-id (or (some-> after-tx-id (+ 1)) 0)}` (I believe that is essentially all the actually needs doing)